app/src/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -1353,7 +1353,12 @@ public int compare(FormIndex arg0, FormIndex arg1) {
                     }
                 }
             } else {
-                Log.w(t, "Unknown view type rendered while current event was question or group! View type: " + mCurrentView == null ? "null" : mCurrentView.getClass().toString());
+                String viewClass = "null";
+                if (mCurrentView != null) {
+                    viewClass = mCurrentView.getClass().toString();
```

Comment:
Fixing a very very bizarre Null Pointer Exception error. 
Turns out conditional branch short circuiting doesn't occur sometimes for Java's turnary operator if auto-boxing/type-casting stuff needs to happen... Nice one Java...